src/public/version.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NINJA_PUBLIC_VERSION_H_
+#define NINJA_PUBLIC_VERSION_H_
+
+namespace ninja {
+/// The version number of the current Ninja release.  This will always
+/// be "git" on trunk.
+extern const char* kNinjaVersion;
```

Comment:
I know that this is already a char* in trunk, but it might be worth doing something else if things are already being done.

```cpp
struct VersionInfo
{
    major
    minor
    patch
    hotfix
    vcs-commit-id
}
```
?

It's not like it's complicated to pre-process the code via the continuous integration system.

Maybe a discussion for another PR though.